Missing Index mapping settings

philkra opened the pull request:
addresses the reported missing index (advanced) mapping settings in https://github.com/elastic/elasticsearch-specification/issues/1499.

delvedor commented on specification/indices/_types/IndexSettings.ts:
```diff
@@ -227,7 +257,43 @@ export class IndexVersioning {
+   * See Shard allocation for shrink.
+   * @server_default 12h
+   */
+  'step.wait_time_threshold'?: Time
```
Are we sure that this field is **always** a "dotted" field? if it can also be represented as a nested field, then we should represent it as such.

delvedor commented on specification/indices/_types/IndexSettings.ts:
```diff
@@ -269,3 +383,119 @@ export class Queries {
+
+/**
+ * Mapping Limit Settings
+ * @doc_url https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-settings-limit.html
```
Can you create a `doc_id` entry for this?
https://github.com/elastic/elasticsearch-specification/blob/main/specification/_doc_ids/table.csv

delvedor commented on specification/indices/_types/IndexSettings.ts:
```diff
@@ -269,3 +383,119 @@ export class Queries {
+   * degradations and memory issues, especially in clusters with a high load or few resources.
+   * @server_default 1000
+   */
+  'total_fields.limit'?: integer
```
Same "dotted" field question as above.

delvedor commented on specification/indices/_types/IndexSettings.ts:
```diff
@@ -269,3 +383,119 @@ export class Queries {
+  /**
+   * The indexing slow log, similar in functionality to the search slow log. The log file name ends with `_index_indexing_slowlog.json`.
+   * Log and the thresholds are configured in the same way as the search slowlog.
+   * @doc_url https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-slowlog.html#index-slow-log
```
Can you create a `doc_id` entry for this?
https://github.com/elastic/elasticsearch-specification/blob/main/specification/_doc_ids/table.csv

delvedor commented on specification/indices/_types/IndexSettings.ts:
```diff
@@ -269,3 +383,119 @@ export class Queries {
+   * the node will reject new coordinating and primary operations. When replica operations consume 1.5x this limit,
+   * the node will reject new replica operations. Defaults to 10% of the heap.
+   */
+  'memory.limit'?: integer
```
Same "dotted" field question as above.

delvedor reviewed (changes requested):
Great work! Left a few comments.

philkra commented:
Thank you @delvedor for the review, I addressed all of your feedback in https://github.com/elastic/elasticsearch-specification/pull/1500/commits/d21c0b04f6238df06b2ef657dbd6308e8e553fd9

delvedor reviewed (approved):
LGTM
